File: src/test/resources/protocols/main/java/io/vlingo/gradle/actortest/Test1ProtocolActor.java
package io.vlingo.gradle.actortest;

import java.util.List;

import io.vlingo.actors.Actor;

public class Test1ProtocolActor extends Actor implements Test1Protocol {

    private static int doSomethingElseWithValue;
    public static int getDoSomethingElseWithValue() { return doSomethingElseWithValue; }
    public static void setDoSomethingElseWithValue(int value) { doSomethingElseWithValue = value; }

    private static String doSomethingWith;
    public static String getDoSomethingWith() { return doSomethingWith; }
    public static void setDoSomethingWith(String value) { doSomethingWith = value; }

    @Override
    public void doSomethingWith(String name, List<String> texts) {
        final StringBuilder builder = new StringBuilder();

        builder.append(name);

        for (final String text : texts) {
            builder.append(text);
        }

        doSomethingWith = builder.toString();
    }

    @Override
    public void doSomethingElseWith(int value) {
        doSomethingElseWithValue = value;
    }
}

File: src/test/resources/protocols/main/kotlin/io/vlingo/gradle/actortest/Test1ProtocolActor.kt
@file:JvmName("Test1ProtocolActor")
package io.vlingo.gradle.actortest;

import io.vlingo.actors.Actor

class Test1ProtocolActor : Actor(), Test1Protocol {

    companion object {

        private var doSomethingElseWithValue: Int? = null
        @JvmStatic fun getDoSomethingElseWithValue(): Int? = doSomethingElseWithValue
        @JvmStatic fun setDoSomethingElseWithValue(value: Int?) { doSomethingElseWithValue = value }

        private var doSomethingWith: String? = null
        @JvmStatic fun getDoSomethingWith() = doSomethingWith
        @JvmStatic fun setDoSomethingWith(value: String?) { doSomethingWith = value }
    }

    override fun doSomethingWith(name: String, texts: List<String>) {
        val builder = StringBuilder()

        builder.append(name)

        for (text in texts) {
            builder.append(text)
        }

        doSomethingWith = builder.toString();
    }

    override fun doSomethingElseWith(value: Int) {
        doSomethingElseWithValue = value
    }
}
